src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/jpa/NonAssociation.kt

```diff
@@ -74,4 +77,20 @@ data class NonAssociation(
       authorisedBy = authorisedBy ?: "",
     )
   }
+
+  override fun equals(other: Any?): Boolean {
+    if (this === other) return true
+    if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
+
+    other as NonAssociation
+
+    if (firstPrisonerNumber != other.firstPrisonerNumber) return false
+    return secondPrisonerNumber == other.secondPrisonerNumber
+  }
+
+  override fun hashCode(): Int {
+    var result = firstPrisonerNumber.hashCode()
+    result = 31 * result + secondPrisonerNumber.hashCode()
+    return result
+  }
```

Comment:
Just out of curiosity, what are these methods for? Are they needed for some of the functionality that was added or more like something that could be useful in the future?

Reply:
These are very JPA additions and are often need when dealing with collections